robot/client/client.go

```diff
@@ -612,8 +613,16 @@ func (rc *RobotClient) createClient(name resource.Name) (resource.Resource, erro
 }
 
 func (rc *RobotClient) resources(ctx context.Context) ([]resource.Name, []resource.RPCAPI, error) {
-	ctx, cancel := contextutils.ContextWithTimeoutIfNoDeadline(ctx, defaultResourcesTimeout)
-	defer cancel()
+	// RSDK-5356 If we are in a testing environment, never apply
+	// defaultResourcesTimeout. Tests run in parallel, and if execution of a test
+	// pauses for longer than 5s, below calls to ResourceNames or
+	// ResourceRPCSubtypes can result in context errors that appear in client.New
+	// and remote logic.
+	if flag.Lookup("test.v") == nil {
```

Comment:
Since we're on go 1.19, I don't have access to [`testing.Testing()`](https://pkg.go.dev/testing#Testing). This seems to be the accepted workaround for pre-go-1.21. It apparently [does not work](https://github.com/golang/go/issues/52600#issuecomment-1489733949) on go 1.21, and we will be forced to switch to `testing.Testing()`.

Reply:
We should probably have this note about needing to use `testing.Testing()` once we upgrade to go 1.21 in the comment as well.

Reply:
Good point; left a `TODO` pointing to APP-2917 which should upgrade us to 1.21.

Reply:
I'm on board as this being the right solution and path forward. Nice job on the TODO.